test/others/config-file/run.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# This script tries to run criu with different options to ensure
+# the configuration file and option handling does not break.
+#
+# The options have been selected by looking at the existing
+# test code coverage and missing code coverage should be handled
+# by the options used in this file.
+#
+# This script tries to only exit if criu crashes. A return value
+# of '1' should not stop the script.
+
+set -xbm
+
+#shellcheck disable=SC1091
+source ../env.sh
+
+if [ ! -d /etc/criu ]; then
+	mkdir -p /etc/criu
+fi
+
+if [ ! -e /etc/criu/default.conf ]; then
+	touch /etc/criu/default.conf
+fi
+
+# This tries to capture any exit codes other than 0 and 1
```

Comment:
We probably need to think about how to implement unit tests that will check correctness.

Reply:
Yes, I was just looking at code coverage to see which part of the code was untested and tried to use all those options which code path was untested.

Reply:
Do you have experience with unit test C code? Any frameworks you know?

Reply:
> Do you have experience with unit test C code? Any frameworks you know?

Hey,

Maybe the [CMocka](https://cmocka.org/) will useful?

Reply:
Let's try without any frameworks and then we will see whether we need something or not. You can create a test binary that uses our config parser. It can read configs from a specified directory, parse them and check that results are correct.

Reply:
I tried to create a unit test. Please have a look. Our code is very interdependent so it is not easy to take out small parts for unit testing, but it worked for the configuration file parser.